doc/user_guide/io.rst

```diff
@@ -1030,6 +1032,162 @@ HDF5 file. All other keyword arguments will be passed to
+one or more datasets. (default is ``None`` all Nexus Metadata will be read)
+ 
+Reading the sole Nexus dataset within a Nexus file:
+
+.. code-block:: python
+
+    >>> hs.load("sample.nxs")
+
+If multiple datasets are present within the Nexus file and you try the same 
+command again **all** available NXdata datasets will be loaded.
+To load NXdata and hdf datasets as well
+
+.. code-block:: python
+
+    >>> hs.load("sample.nxs",nxdata_only=False)
+
+.. note::
+
+    Given that HDF5 files can accommodate very large datasets, setting 
+    ``lazy=True`` is strongly recommended if the contents of the HDF5 file 
+    are not known apriori. This prevents issues with regard to loading datasets
+    far larger than memory.
+
+    Also note that setting ``lazy=True`` leaves the file handle to the 
+    HDF5 file open.
+    If it is important that the files be closed after reading,
+    set ``lazy=False``.
+
+.. code-block:: python
+
+    >>> hs.load("sample.h5")
+    [<Signal2D, title: HAADF, dimensions: (|128, 128)>,
+    <Signal1D, title: EELS, dimensions: (|64, 64, 1024)>]
+
+We can load a specific datasets using the ``dset_search_path`` keyword argument. 
+Setting it to the absolute path of the desired dataset will cause 
+the single dataset to be loaded.
+Setting it to a substring or list of substring, datasets who's paths contain 
+one of the substrings will be loaded.
+
+
+.. code-block:: python
+
+    >>> # Loading a specific dataset
+    >>> hs.load("sample.h5", dset_search_path='/entry/experiment/EDS/data')
+
+We can also choose to load only specific items from the metadata using
+ ``dset_path`` keyword argument. This can be a list of 
+absolute paths of the desired dataset or substrings to find in the metadata
+path.
+
+.. code-block:: python
+
+    >>> # Loading a specific dataset
+    >>> hs.load("sample.h5", dset_search_path='EDS')
+
+Nexus files also support parameters or dimensions that have been varied 
+non-linearly.
+Since HyperSpy Signals expect linear variation of parameters / axes, such 
+non-linear information would be lost in the axes manager and replaced with
+indices
+
+
+Writing
+^^^^^^^
+Signals can be written to new Nexus files using the standard ``.save()`` 
+function.
+When Hyperspy creates a signal it creates and holds Hyperspy metadata 
+and original_metadata seperately.
+When we load a Nexus file the Nexus metadata is stored in 
+original_metadata. Conversely when we save to Nexus only the 
+original_metadata is currently stored to the Nexus file along with 
+the signal data and axes. 
+
+.. code-block:: python
+
+    >>> sig.save("output.nxs")
+
+If you wish to save using the ``.hdf5``.``.h5`` or ``.hdf`` extensions
+you must specify the format as Nexus
+
+.. code-block:: python
+
+    >>> sig.save("output.h5",preferred_format="Nexus")
+
+Note that the model and other secondary data artifacts linked to the signal 
+are not written to the file but these can be implemented at a later stage.
```

Comment:
I'd just remove the "but these can be implemented at a later stage" from the docs and, assuming this is merged without that, we can raise an issue to say that it should be implemented eventually.

Reply:
Changed the documentation quite a bit. Took your comment on being pragmatic about what works for us so made changes to add saving hyperspy metadata and several signals based on workflows. Also clarified the differences in how metadata is structured to match the hdf and nexus formats - e.g. attributes with datasets